runtime/middleware/jwt.go

```diff
@@ -0,0 +1,95 @@
+package middleware
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/hypermodeinc/modus/runtime/config"
+	"github.com/hypermodeinc/modus/runtime/logger"
+	"github.com/hypermodeinc/modus/runtime/utils"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type JWTClaimsKey string
+
+const JWTClaims JWTClaimsKey = "jwt_claims"
+
+func HandleJWT(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var ctx context.Context = r.Context()
+		tokenStr := r.Header.Get("Authorization")
+		tokenStr = strings.TrimPrefix(tokenStr, "Bearer ")
+
+		privKeysStr := os.Getenv("MODUS_RSA_PEMS")
+		if privKeysStr == "" {
+			if !config.IsDevEnvironment() || tokenStr == "" {
+				next.ServeHTTP(w, r)
+				return
+			}
+			token, _, err := new(jwt.Parser).ParseUnverified(tokenStr, jwt.MapClaims{})
+			if err != nil {
+				logger.Debug(r.Context()).Err(err).Msg("JWT parse error")
+				next.ServeHTTP(w, r)
+				return
+			}
+			if claims, ok := token.Claims.(jwt.MapClaims); ok {
+				ctx = AddClaimsToContext(ctx, claims)
+			}
+			next.ServeHTTP(w, r.WithContext(ctx))
+			return
+
+		}
+
+		var privKeysUnmarshalled map[string]string
+		err := json.Unmarshal([]byte(privKeysStr), &privKeysUnmarshalled)
+		if err != nil {
+			logger.Error(r.Context()).Err(err).Msg("JWT private keys unmarshalling error")
+			http.Error(w, "Internal server error", http.StatusInternalServerError)
+			return
+		}
```

Comment:
This is in the hot path of every request. We should instead be reading and parsing the keys just once on startup.

Please move the logic for reading the env var, unmarshalling it, and also the parsing logic for the keys themselves, into an `Init` function, in this file.

Ultimately when `jwt.Parse` is called, it should only have to parse the incoming JWT from the request, and everything else should be set up in advance.